odoo/http.py

```diff
@@ -985,6 +985,28 @@ def generate_key(self, salt=None):
     def is_valid_key(self, key):
         return _base64_urlsafe_re.match(key) is not None
 
+    def get_missing_session_identifiers(self, identifiers):
+        """
+            :param identifiers (iterable): session identifiers whose file existence must be checked
+            :return (set): the identifiers which are not present on the filesystem
+        """
+        # There are a lot of session files.
+        # Use the param ``identifiers`` to select the necessary directories.
+        # In the worst case, we have 4096 directories (64^2).
+        identifiers = set(identifiers)
+        directories = {
+            os.path.normpath(os.path.join(self.path, identifier[:2]))
+            for identifier in identifiers
+        }
+        # Get the list of session files for the directories.
+        # Only the first 42 characters are needed.
+        session_file_list = []
+        for directory in directories:
+            with contextlib.suppress(OSError), os.scandir(directory) as session_files:
+                session_file_list.extend(sf.name[:42] for sf in session_files)
+        # Remove the identifiers for which a file is present on the filesystem.
+        return identifiers - set(session_file_list)
```

Comment:
I think this uses less memory and takes about the same time.
```suggestion
        # Remove the identifiers for which a file is present on the filesystem.
        for directory in directories:
            with contextlib.suppress(OSError), os.scandir(directory) as session_files:
                identifiers.difference_update(sf.name[:42] for sf in session_files)
        return identifiers
```
That's because we make a mutable copy higher up at `identifiers = set(identifiers)`